WP3_Advanced_RAG_discussion.md

```diff
@@ -107,14 +107,56 @@ E-->G[Retrieved content]
 This technique requires an LLM to produce a potential document, from which the answer to the question could be drawn, and then use that as a semantic tool.
 Other techniques get the LLM to try and answer the question without context, and then use that answer to improve the generation or retrieval. 
 
+Retrieval can also be improved through fine tuning of the embedder models. 
+A specialised embedder, fine tuned on the kind of documents or questions expected in the task, can produce a much more precise vector space and improve retrieval results. 
+
 
 ### Augmentation
+Once the materials have been retrieved, the original query can be augmented with extra context. 
+LLMs perform quite differently depending on how the prompt is worded. Increasingly the task of prompt engineering is being automated and tackled by LLMs themselves. 
+
 Re-Rank
+We know that LLMs are better at retrieving facts from context. More specifically, they have been found to be better at retrieving facts from the start and end of context. 
+One of the simpler re-rank ideas is to place the most relevant retrieved materials at the start and end of the augmented prompt. 
 
```

Comment:
This section is clearly still be added to (great start!) - but some other ideas might be to pass the augmented prompt through a LLM to generate a "standard alone prompt" which combines it all together into something more succinct, which can then be used for generation - I can't remember the name of this technique though... some form of prompt consolidation.